.github/workflows/preview-docs.yml

```diff
@@ -1,7 +1,7 @@
 name: Preview docs
 
 on:
-  pull_request:
+  pull_request_target:
```

Comment:
With `pull_request_target`, this workflow is executed on the PR base branch (= master in most cases), not on the PR branch (e.g. `harupy:test-foo`).